### PR TITLE
feat(auto_moderation): add join and mention raid fields

### PR DESCRIPTION
**Pull request by spifory**
## Summary

- discord/discord-api-docs#5778

## This is a **Code Change**

- [ ] I have tested my changes.
- [x] I have updated the documentation to reflect the changes.
- [x] I have run `task pyright` and fixed the relevant issues.

### CODE REVIEW

File: nextcord/guild.py [hunk 1]
```diff
@@ -520,6 +522,8 @@ def _from_data(self, guild: GuildPayload) -> None:
             "premium_progress_bar_enabled"
         )
 
+        self.safety_alerts_channel_id: Optional[int] = guild.get("safety_alerts_channel_id")
```
Comment (EmreTech) [inline nextcord/guild.py:525]: There should be a property, `safety_alerts_channel`, that returns the underlying channel object this id represents. Similar to `system_channel`, `rules_channel`, and `public_updates_channel`.

File: nextcord/types/auto_moderation.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ class AutoModerationTriggerMetadata(TypedDict, total=False):
     presets: List[KeywordPresetType]
     allow_list: List[str]
     mention_total_limit: Optional[int]
+    mention_raid_protection_enabled: bool
```
Comment (EmreTech) [inline nextcord/types/auto_moderation.py:21]: `nextcord.AutoModerationTriggerMetadata` needs to be updated to handle this attribute.

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: nextcord/guild.py (hunk 1 above)
Comment (ooliver1) [reply nextcord/guild.py:525]: @spifory did you forget to push your commits? This was marked as resolved without anything.

Even if this was kept as an attribute, theres no documentation.

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: nextcord/guild.py (hunk 1 above)
Comment (spifory) [reply nextcord/guild.py:525]: Yep I had forgotten to push my commits, I've pushed them now.

File: nextcord/guild.py [hunk 3]
```diff
@@ -796,6 +802,19 @@ def public_updates_channel(self) -> Optional[TextChannel]:
         channel_id = self._public_updates_channel_id
         return channel_id and self._channels.get(channel_id)  # type: ignore
 
+    @property
+    def safety_alerts_channel(self) -> Optional[TextChannel]:
+        """Optional[:class:`TextChannel`]: Returns the guild's safety alerts channel where admins and
+        moderators of the guild receive notices about DM or join activity that exceeds usual numbers
+        for the guild.
```
Comment (EmreTech) [inline nextcord/guild.py:809]:
```suggestion
        """Optional[:class:`TextChannel`]: Returns the guild's channel where admins and
        moderators of the guild receive safety alerts from Discord. The guild must be a 
        Community guild.
```
This is a little more broad, which is good if and when the safety alerts channel gets different alerts in the future.

**Review by EmreTech — CHANGES_REQUESTED** [review]

**Review by EmreTech — APPROVED** [review]
Tested.

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**